src/t_zset.c

```diff
@@ -2918,7 +2918,9 @@ static void zrangeResultEmitCBufferToClient(zrange_result_handler *handler,
     addReplyBulkCBuffer(handler->client, value, value_length_in_bytes);
 
     if (handler->withscores) {
-        addReplyDouble(handler->client, score);
+        char scorebuf[MAX_D2STRING_CHARS];
+        int scorelen = d2string(scorebuf,sizeof(scorebuf),score);
+        addReplyBulkCBuffer(handler->client, scorebuf, scorelen);
```

Comment:
doesn't this violate RESP3 and a few other things addReplyDouble does?
seems that there's no test (one that uses `readraw`) for this (checking RESP3).

how about adding the `double2ll` trick to addReplyDouble instead?

also, maybe d2string should be using fpconv_dtoa instead of sprintf?

Reply:
@oranagra  WRT
 
> also, maybe d2string should be using fpconv_dtoa instead of sprintf?

it's already using using it. So I guess we can simplify addReplyDouble instead to use d2string.